pkg/apis/keda/v1alpha1/triggerauthentication_types.go

```diff
@@ -83,6 +92,157 @@ type AuthEnvironment struct {
 	ContainerName string `json:"containerName"`
 }
 
+// Vault is used to authenticate using Hashicorp Vault
+// +k8s:openapi-gen=true
+type Vault struct {
+	Address        string              `json:"address"`
+	Authentication VaultAuthentication `json:"authentication"`
+
+	// +listType
+	Secrets []VaultSecret `json:"secrets"`
+
+	// +optional
+	Credential Credential `json:"credetial"`
+
+	// +optional
+	Role string `json:"role"`
+
+	// +optional
+	Mount string `json:"mount"`
+}
+
+// Credential defines the Hashicorp Vault credentials depending on the authentication method
+// +k8s:openapi-gen=true
+type Credential struct {
+	// +optional
+	Token string `json:"token"`
+
+	// +optional
+	ServiceAccount string `json:"serviceAccount"`
+}
+
+// VaultAuthentication contains the list of Hashicorp Vault authentication methods
+type VaultAuthentication string
+
+// Client authenticating to Vault
+const (
+	VaultAuthenticationToken      VaultAuthentication = "token"
+	VaultAuthenticationKubernetes                     = "kubernetes"
+	// VaultAuthenticationAWS                            = "aws"
+)
+
+// VaultSecret defines the mapping between the path of the secret in Vault to the parameter
+// +k8s:openapi-gen=true
+type VaultSecret struct {
+	Parameter string `json:"parameter"`
+	Path      string `json:"path"`
+	Key       string `json:"key"`
+}
+
 func init() {
 	SchemeBuilder.Register(&TriggerAuthentication{}, &TriggerAuthenticationList{})
 }
+
+// Authenticate returns a client connected to Vault
+func (v *Vault) Authenticate(logger logr.Logger) (*vaultApi.Client, error) {
```

Comment:
I wonder whether `pkg/apis/keda/v1alpha1/triggerauthentication_types.go` is the right place for `func` definitions

Reply:
Do you think I should create a new Hashicorp Vault handler and put it there?

Reply:
created the hashicorpvault_handler.go file with all the func definitions